### PR TITLE
Simplify Pmd custom rules configuration

### PR DESCRIPTION
**Pull request by 6hundreds**

### CODE REVIEW

**Comment by 6hundreds** [comment]
@bot-gradle test this

**Comment by 6hundreds** [comment]
@bot-gradle test this

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdExtension.java [hunk 1]
```diff
@@ -34,33 +36,37 @@ public class PmdExtension extends CodeQualityExtension {
...
+        this.incrementalAnalysis = project.getObjects().property(Boolean.class);
+        this.maxFailures = project.getObjects().property(Integer.class);
+        this.threads = project.getObjects().property(Integer.class);
+        this.ruleSets = project.getObjects().listProperty(String.class);
     }
 
     /**
      * The built-in rule sets to be used. See the <a href="https://pmd.github.io/pmd-6.39.0/pmd_rules_java.html">official list</a> of built-in rule sets.
      *
+     * If not configured explicitly, the returned conventional value is "category/java/errorprone.xml", unless {@link #getRuleSetConfig()} returns
```
Comment (donat) [inline subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdExtension.java:61]:
```suggestion
     * If not configured explicitly, the returned conventional value is "category/java/errorprone.xml", unless {@link #getRuleSetConfig()} returns.
```

Missing dot at the end of the sentence.

**Review by donat — APPROVED** [review]
LGTM. Let's wait for @nathan-contino to review the documentation in this PR.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 2]
```diff
@@ -171,6 +171,11 @@ eclipse {
 }
 ```
 
+==== Simplifying PMD custom rules configuration
+
+Previously to configure PMD to use custom rules and ignore the default ones you have to explicitly clear default rules with following line `ruleSets = []`.
+In the Gradle 8.0 it's done by default as soon as `ruleSetConfig` or `ruleSetFiles` was set to non-empty value.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:177]:
```suggestion
Previously, you had to explicitly configure PMD to ignore default rules with `ruleSets = []`.
In the Gradle 8.0, setting `ruleSetConfig` or `ruleSetFiles` to a non-empty value implicitly ignores default rules.
```
Suggestion: trim some words, wording consistency.
Consider defining the PMD acronym in the first sentence with "Pxxxx Mxxxx Dxxxxx (PMD)", and possibly provide a link to other documentation on the subject.

**Review by nathan-contino — COMMENTED** [review]
Suggested a docs wording improvement and a couple of clarifications.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 2 above)
Comment (6hundreds) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: @nathan-contino Thanks!  Funny thing is PMD documentation doesn't clarifying what PMD acronym stands for:)

**Comment by 6hundreds** [comment]
@bot-gradle test and merge

**Pull request merged**